docs/src/reference/architecture.md

```diff
@@ -0,0 +1,72 @@
+
+A set of external systems need to be easily integrated with our k8s distribution. We have identified the following:
+ - **Load-balancer**: Although the k8s snap distribution comes with a load-balancer we expect the end customer environment to have a load-balancer and thus we need to integrate with it.
+- **Storage**: Kubernetes typically expects storage to be external to the cluster. The k8s snap comes with a local storage option but we still need to offer proper integration with any storage solution.
+- **Identity management**: Out of the box the k8s snap offers credentials for an admin user. The admin user can complete the integration with any identity management system available or do user management manually.
+- **External datastore**: By default, Kubernetes uses etcd to keep track of state. Our k8s snap comes with `dqlite` as its datastore. We should however be able to use any end client owned datastore installation. That should include an external `postgresql` or `etcd`.
+
+## The k8s snap
+
+Looking more closely at what is conatined within the K8s snap istelf:
+
+```{kroki} ../assets/k8s-container.puml
+```
+
+The k8s snap distribution includes the following:
+
+- **Kubectl**: through which users and other systems interact with  Kubernetes and drive the cluster operations.
+- **K8s upstream services**: These are Kubernetes binaries built from upstream and shipped in the snap.
+- **Components** are the workloads and features we deem important to be available to our users and therefore are shipped in the snap and are enabled, configured and disabled in a guided way.
+- State is backed up by **dqlite** by default, which keeps that state of the Kubernetes cluster as well as the the state we maintain for the needs of the cluster operations.
+- **Runtime**: `containerd` and `runc` are the shipped container runtimes.
+- **K8sd**: which implements the operations logic and exposes that functionality via CLIs and REST APIs.
+
+## K8sd
+
+K8sd is the component that implements and exposes the operations functionality needed for managing the Kubernetes cluster.
+
+```{kroki} ../assets/k8sd-component.puml
+```
+
+At the core of the `k8sd` functionality we have the components and cluster managers:
+The components manager is responsible for the workload features we deem important for a Kubernetes cluster. Namely:
+
+- DNS
+- CNI
+- ingress
+- gateway API
+- load-balancer
+- local storage
+- observability
+
+The cluster manager is responsible for implementing the formation of the cluster. This includes operations such as joining/removing nodes into the cluster and reporting status.
+
+This functionality is exposed via the following interfaces:
+
+- The **CLI**: The CLI is available to only the root user on the k8s snap and all CLI commands are mapped to respective REST calls.
+
+- The **REST API**: apart from serving the CLI, is also used by the charm that has to programmatically drive the Kubernetes cluster. 
```

Comment:
We provide a `JSON RPC API` to be specific. 
However, people might be more familiar with the `REST API` term even though it is not 100% correct.

Reply:
Discussed this with @ktsakalozos offline: We do a wild mix of REST and JSON RPC and should just not use those terms at all. However, we still need to mention the protocol.

So something like `HTTP API` or `we provide an API over/via HTTP` should be used instead.